### DISCUSSION TITLE
Some context menu parts are not translatable

### DISCUSSION THREAD

**Discussion by Stheel83** [sqlitebrowser/sqlitebrowser#3532] [General]
Hi

there are some untranslatable context menu parts in every language i tested:

1. in "Edit Database Cell" open the context menu:

    "Copy" and "Select All" are not translatable
![Copy_Select](https://github.com/sqlitebrowser/sqlitebrowser/assets/153910911/8aabacd9-b9e8-48bc-bc14-3f927b87315d)

2. also in the "Execute SQL" tab open a SQL file. Then open the context menu inside the SQL Statement:

    "Undo", "Redo", "Cut", "Copy", "Paste",  "Delete", "Select All" are not translatable
![paste](https://github.com/sqlitebrowser/sqlitebrowser/assets/153910911/b1eb2ede-e6a7-407f-a815-6bf779acac3d)

Looks like these are missing in the ExtendedScintilla.cpp, because "Find and Replace...", in the same context menu, is translatable.
But i can be completely wrong as i don't know how this really works.

**Comment by mgrojo** [comment]
That's probably because the Qt installation of your platform lacks translation for those strings. If this is still happening with the 3.13 release, please, give details about your platform and Qt version, and we can see if there's something we can improve.

**Reply by Stheel83** [reply]
Hi, thanks for the reply.

I checked the missing translations now with the 3.13 version. But still the same.

I tried it under Win 10, Win 11 and Opensuse Leap 15.5.

Under Windows I tried the *.msi and the *.zip install files.

And with Opensuse i used the appimage.

Then i also switched the languages to spain and french bit still no change and also english.

To be clear: the missing translations show only in "edit database cell" and "execute sql". The other context menues translate "save", "copy" etc. into german without problems.

So, still no change.
Additionaly i will check my qt versions the next days

Regards

**Reply by mgrojo** [reply]
Ah, I understand now. That is the Scintilla widget, an external library we use for the code editors, including the "edit database cell" editor, except for RTL languages. I think these are the only visible strings introduced by that library. But the problem is that if we process its files for translators, we get a lot of new strings that aren't visible in our application, and it's a load of work for translators for nothing.

The question is that I don't find those strings inside the library source code itself, so not sure what is happening, in any case.

**Reply by mgrojo** [reply]
Probably it's some interaction between Qt and Scintilla.

**Reply by Stheel83** [reply]
Ok, so then i don't need to post my Qt version anymore, i think.
And i think a can not do anything more right know.

**Comment by hchiper** [comment]
I found the files `sqlitebrowser/libs/qscintilla/Qt4Qt5/qscintilla_*.ts`. 
They have a `QsciScintilla` context, where exactly those 7 mentioned strings are present.
The corresponding `.qm` files are also present.

In qscintilla_de.ts` they are translated. If it is the right file to have the German translation working, then it seems that it is not taken into account. Or maybe I am totally wrong. What do you think?

I also noticed that in `qscintilla_fr.ts` these strings are not translated. Moreover, its 3rd line is `<TS version="2.1" language="en_GB">` but should have `"fr"` instead. Therefore, when opened in Qt Linguist, we are asked for "Translation for British English (United Kingdom)". This is a tiny work of translation and I am willing to do for French if it is useful. 

But if the `qscintilla_*.ts` are auto-generated, it would be preferable to fix the `fr` problem first. Otherwise I can change that with a simple text editor.

**Reply by Stheel83** [reply]
Ok, thanks for search and the informations, so i think i can not do anymore right know for the german translation.
